### PR TITLE
[refactor] Enhance NaN detection in sampling layers

### PR DESCRIPTION
**Pull request by jimmy-evo**
Hi from [novita.ai](https://novita.ai/) team 👋

## Motivation

Dont crash when nan is detected

## Modifications

- Moved NaN detection logic to a dedicated function in common utils.
- Updated sampling layers to utilize the new detect_nan function for improved readability and maintainability.
- Removed redundant NaN checks and warnings from individual layers, centralizing the logic.

results:
```
[2025-12-03 15:29:24 TP0] Detected errors during sampling! NaN in the logits.
[2025-12-03 15:29:24 TP7] Detected errors during sampling! NaN in the logits.
[2025-12-03 15:29:24 TP6] Detected errors during sampling! NaN in the logits.
[2025-12-03 15:29:24 TP5] Detected errors during sampling! NaN in the logits.
[2025-12-03 15:29:24 TP1] Detected errors during sampling! NaN in the logits.
[2025-12-03 15:29:24 TP3] Detected errors during sampling! NaN in the logits.
[2025-12-03 15:29:24 TP2] Detected errors during sampling! NaN in the logits.
[2025-12-03 15:29:24 TP4] Detected errors during sampling! NaN in the logits.
```

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [x] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).
- [ ] Work with maintainers to merge your PR. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process)

### CODE REVIEW

**Comment by hnyls2002** [comment]
Thanks @jimmy-evo. NaN detection has since been consolidated on main into a shared utility (`sglang/srt/utils/async_probe.py`, `maybe_detect_nan`) used by the sampler and the speculative workers, and `eagle_worker.py` that this PR patches has been removed in the spec v2 refactor. Closing as superseded; please reopen if I've missed something.
